compiler/test/graindoc/types.input.gr

```diff
@@ -0,0 +1,35 @@
+module TypeGrainDoc
+
+/**
+ * A record
+ */
+provide record R {
+  /**
+   * Record field x
+   */
+  x: Number,
+  /**
+   * Record field y
+   * Second line
+   */
+  y: String,
+}
+
+/**
+ * An enum
+ */
+provide enum E {
+  /**
+   * Enum variant
+   */
+  Variant1,
+  /**
+   * Another variant
+   */
+  Variant2(String, Number),
```

Comment:
Can we get a variant testing `Variant3{ t: Number }` it would be nice if we tested nested docs here?

After looking a little closer as your implementation I dot think you handle allowing docs on nested records we might want to handle that somehow. Idk how just yet maybe we append them to the table like `Variant.t` and the description, though that could be confusing.

Reply:
Yeah I thought about this case as well and am unsure of the best way to display it; including a nested table in a new column is one option, and although I initially discounted this option it may be better than separating it out into another top-level table as that would cause an awkward reading flow imo :thinking:

Reply:
I tried playing around with different formatting styles and I think having inline record data in a nested table seems like an ok option; since there would only ever be one layer of nesting I think it wouldn't be unreasonable. For example the following code could generate this table:
```
provide enum E {
  /**
   * Singleton constr
   */
  S,
  /**
   * Tuple constr
   */
  T(Number),
  /**
   * Record constr
   */
  R {
    /**
     * A number
     */
    x: Number,
    /**
     * A function
     */
    f: Number -> Number
  }
}
```
![image](https://github.com/grain-lang/grain/assets/48104782/9ca4a16c-61b6-480d-b65f-cc086598aa51)

The image was generated with https://markdownlivepreview.com/, but will probably look similar on the website. Thoughts on this display format?

Reply:
It looks better than I was expecting but I wonder if anyone else has any better ideas. Before we implement.

Reply:
I ended up going ahead and implementing it, if there are suggestions for better output styling I'll update